support-workers/src/typescript/services/salesforce.ts

```diff
@@ -88,23 +109,27 @@ export class SalesforceService {
 		user: User,
 		giftRecipient: GiftRecipient | null,
 	): Promise<SalesforceContactRecord> => {
-		console.log('SalesforceService: Creating contact records');
-		const contactRecordRequest = createContactRecordRequest(
-			user,
-			giftRecipient,
-		);
-		// First upsert the buyer contact record
-		const buyerResponse = await this.upsert(contactRecordRequest);
+		const hasGiftRecipient = !!giftRecipient;
+		const buyerType = getBuyerType(user, hasGiftRecipient);
+		const buyerContact = createBuyerRecordRequest(user, buyerType);
+		const buyerResponse = await this.upsert(buyerContact);
 		const giftRecipientResponse = await this.maybeAddGiftRecipient(
 			buyerResponse.ContactRecord,
 			giftRecipient,
 			user,
 		);
-		return giftRecipientResponse?.ContactRecord ?? buyerResponse.ContactRecord;
+		const recipientContactRecord =
+			giftRecipientResponse?.ContactRecord ?? buyerResponse.ContactRecord;
+
+		return recipientContactRecord;
 	};
 
 	upsert = async (
-		contact: ContactRecordRequest | DeliveryContactRecordRequest,
+		contact:
+			| PrintContactRecordRequest
+			| GiftBuyerContactRecordRequest
+			| GiftRecipientContactRecordRequest
+			| DigitalOnlyContactRecordRequest,
```

Comment:
I wonder if binding these together in a sum type (called something like `ContactRecordRequest`) would make sense? Given that we probably have multiple places where any one of these is accepted:

```ts
type ContactRecordRequest = PrintContactRecordRequest | GiftBuyerContactRecordRequest | GiftRecipientContactRecordRequest | DigitalOnlyContactRecordRequest;
```

Reply:
Makes sense, I've added the following types:

```
type BuyerTypeRecordRequest =
	| PrintContactRecordRequest
	| GiftBuyerContactRecordRequest
	| DigitalOnlyContactRecordRequest;
type ContactRecordRequest =
	| BuyerTypeRecordRequest
	| GiftRecipientContactRecordRequest;
	```